<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.550">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leighton Pritchard">

<title>BM214 Workshop 2: Sequence Searching - 1&nbsp; Introduction</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./pairwise.html" rel="next">
<link href="./index.html" rel="prev">
<link href="./icon_32.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./intro.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">BM214 Workshop 2: Sequence Searching</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/sipbs-compbiol/BM214-Workshop-2" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface to 2023-24 BM214 Workshop 2: Sequence Searching</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./pairwise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Pairwise Sequence Alignment</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./blast.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">BLAST: Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scenario-1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Scenario 1: 16S Identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scenario-2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Scenario 2: Protein Identification</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scenario-3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Scenario 3: Annotating a Metagenomic Contig</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./glossary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Glossary</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./resources.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Additional Resources</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#why-do-we-care-about-biological-sequence-similarity" id="toc-why-do-we-care-about-biological-sequence-similarity" class="nav-link active" data-scroll-target="#why-do-we-care-about-biological-sequence-similarity"><span class="header-section-number">1.1</span> Why do we care about biological sequence similarity?</a></li>
  <li><a href="#why-does-sequence-similarity-imply-functional-similarity" id="toc-why-does-sequence-similarity-imply-functional-similarity" class="nav-link" data-scroll-target="#why-does-sequence-similarity-imply-functional-similarity"><span class="header-section-number">1.2</span> Why does sequence similarity imply functional similarity?</a></li>
  <li><a href="#how-similar-is-similar-enough" id="toc-how-similar-is-similar-enough" class="nav-link" data-scroll-target="#how-similar-is-similar-enough"><span class="header-section-number">1.3</span> How similar is similar enough?</a></li>
  <li><a href="#where-can-we-go-to-find-similar-sequences" id="toc-where-can-we-go-to-find-similar-sequences" class="nav-link" data-scroll-target="#where-can-we-go-to-find-similar-sequences"><span class="header-section-number">1.4</span> Where can we go to find similar sequences?</a>
  <ul class="collapse">
  <li><a href="#insdc" id="toc-insdc" class="nav-link" data-scroll-target="#insdc"><span class="header-section-number">1.4.1</span> INSDC</a></li>
  <li><a href="#genbank" id="toc-genbank" class="nav-link" data-scroll-target="#genbank"><span class="header-section-number">1.4.2</span> GenBank</a></li>
  <li><a href="#refseq" id="toc-refseq" class="nav-link" data-scroll-target="#refseq"><span class="header-section-number">1.4.3</span> RefSeq</a></li>
  <li><a href="#uniprot" id="toc-uniprot" class="nav-link" data-scroll-target="#uniprot"><span class="header-section-number">1.4.4</span> UniProt</a></li>
  <li><a href="#interpro" id="toc-interpro" class="nav-link" data-scroll-target="#interpro"><span class="header-section-number">1.4.5</span> InterPro</a></li>
  </ul></li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><span class="header-section-number">1.5</span> Summary</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sipbs-compbiol/BM214-Workshop-2/edit/main/intro.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM214-Workshop-2/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<blockquote class="blockquote">
<p>If two biological sequences are similar enough to each other, then we can hypothesise that they - or their products, in the case of coding sequences - may derive from a recent common ancestor, perform the same biological function, or adopt the same structural conformation.</p>
</blockquote>
<section id="why-do-we-care-about-biological-sequence-similarity" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="why-do-we-care-about-biological-sequence-similarity"><span class="header-section-number">1.1</span> Why do we care about biological sequence similarity?</h2>
<p>It may be surprising, but we do not know what all genes in an organism do. In some organisms, we may not even know what <em>most</em> genes do, or even whether we know all the genes that are present.</p>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>Model organisms, such as <em>Escherichia coli</em>, are possibly the forms of life we understand best. But even the <a href="https://www.ncbi.nlm.nih.gov/nuccore/NC_002695.2">genome of <em>E. coli</em> O157:H7</a> - maybe the best-understood genome we know - contains 756 coding sequences (CDS) annotated as <code>hypothetical protein</code>, which means we have no evidence of what their functions are. This is 756/5067 ≈ 15% of the predicted proteome for that organism.</p>
</div>
</div>
<p>As biological sequences diverge through the actions of random mutation and nonrandom natural selection, they accumulate differences between them. The smaller the amount of time since two sequences diverged, the less time they have to accumulate these changes and, as a result, sequences from more closely-related organisms tend to be more similar to each other.</p>
<p>In general, the more similar two protein sequences are, the more likely they are to share the same functional activity, and to perform the same job for the organism they are found in. So, if we have a biological sequence of unknown function, and we can find a protein whose function we know whose sequence is similar <em>enough</em> to our sequence, then we can hypothesise that they have the same function.</p>
<p>Part of the reason the sequence of a protein enables its function is that the sequence dictates how the protein <em>folds</em> into its final shape. This shape places catalytic centres, and recognition surfaces for other proteins and small molecules, into the right place for the function to happen. Similar protein sequences fold up into similar shapes.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Functional Annotation Transfer
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Assigning a proposed function, on the basis of sequence similarity to another sequence that has an annotated function, is called <strong>functional annotation transfer</strong>. The vast majority of genes and gene products in public databases have functions assigned by this approach, <em>not</em> by direct experimental verification. New sequences are added to these databases at a rate that far exceeds our ability to characterise their products experimentally, so this situation is likely to persist for some time.</p>
<p>The <a href="https://biofunctionprediction.org/cafa/">Critical Assessment of Functional Annotation</a> (CAFA) is series of experiments and challenges designed to understand how well our computational methods of functional annotation match experimental reality (<span class="citation" data-cites="Zhou2019-ny">Zhou et al. (<a href="references.html#ref-Zhou2019-ny" role="doc-biblioref">2019</a>)</span>).</p>
</div>
</div>
</div>
</section>
<section id="why-does-sequence-similarity-imply-functional-similarity" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="why-does-sequence-similarity-imply-functional-similarity"><span class="header-section-number">1.2</span> Why does sequence similarity imply functional similarity?</h2>
<p>If two proteins have the exact same sequence of amino acids, then they are essentially “the same thing.” We would expect them both to fold into the same structure, and to interact with the same molecules in the same ways. But the more changes we make to one of the sequences, the more likely it is that we will introduce a difference that results in a change to the structure or the functional activity of the molecule.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
It is not true that the number of sequence differences between two proteins is <em>proportional to</em> the difference in function.
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>For example, the activity of an enzyme like chymotrypsin relies on the exact placement of three amino acid residues in a <em>catalytic triad</em> (<a href="#fig-chymotrypsin" class="quarto-xref">Figure&nbsp;<span>1.1</span></a>). If any one of these residues is substituted inappropriately, the enzyme no longer acts to catalyse the reaction that processes its substrate. However, several simultaneous substitutions can be tolerated elsewhere on the protein structure with no significant loss of catalytic function.</p>
<div id="fig-chymotrypsin" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-chymotrypsin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/1cbw.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chymotrypsin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.1: The <a href="https://www.rcsb.org/structure/1CBW">1CBW</a> structure of bovine chymptrypsin (transparent space-filling model), complexed with the inhibitor BPTI (rainbow-coloured cartoon). The catalytic triad Asp102-His57-Ser195 is shown in ball-and-stick representation. Changing any member of the catalytic triad for another residue would prevent chymotrypsin from working as an enzyme.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="how-similar-is-similar-enough" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="how-similar-is-similar-enough"><span class="header-section-number">1.3</span> How similar is similar enough?</h2>
<p>There is no single value of “similar” short of 100% exact identity that is guaranteed to be “similar enough” to imply identical function. What might qualify as “similar enough” depends on the scientific question you are asking, and will vary if you are asking questions like:</p>
<ul>
<li>Are these two biological sequences from the same organism?</li>
<li>Do these two biological sequences share a single ancestor?</li>
<li>Do these two enzymes share the same catalytic function?</li>
<li>Do these two enzymes catalyse the same reactions for the same substrates?</li>
<li>Do these two proteins fold into the same <span class="math inline">\(\alpha\)</span>-carbon backbone structure?</li>
</ul>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>There are some <em>heuristics</em> (rules of thumb) that can guide you in how confident you should be in, say, assigning functional annotation or common structure on the basis of sequence identity. The table in <a href="#fig-similarity-table" class="quarto-xref">Figure&nbsp;<span>1.2</span></a> (from <span class="citation" data-cites="Punta2008-gq">Punta and Ofran (<a href="references.html#ref-Punta2008-gq" role="doc-biblioref">2008</a>)</span>) outlines some guidelines for interpreting functional similarity from sequence identity.</p>
<div id="fig-similarity-table" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-similarity-table-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/pcbi.1000160.t001.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-similarity-table-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.2: Table relating protein sequence identity to expected functional similarity at a number of levels, reproduced from (<span class="citation" data-cites="Punta2008-gq">Punta and Ofran (<a href="references.html#ref-Punta2008-gq" role="doc-biblioref">2008</a>)</span>
</figcaption>
</figure>
</div>
<p>Similarly, we can often consider proteins sharing at least 40% amino acid identity to also have a similar <span class="math inline">\(\alpha\)</span>-carbon backbone structure. Proteins sharing less than 20% identity are likely to have quite dissimilar structures, and the range 20-40% identity is considered a <em>twilight zone</em> where it is difficult to predict whether two proteins have a common structure (<a href="#fig-rmsd_vs_identity" class="quarto-xref">Figure&nbsp;<span>1.3</span></a>).</p>
<div id="fig-rmsd_vs_identity" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-rmsd_vs_identity-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/bioinformatics_29_13_1589_f1.jpeg" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rmsd_vs_identity-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.3: A plot of pairwise protein structure dissimilarity (rmsd) against pairwise protein sequence identity for a set of membrane-spanning proteins, reproduced from <span class="citation" data-cites="Olivella2013-rm">Olivella et al. (<a href="references.html#ref-Olivella2013-rm" role="doc-biblioref">2013</a>)</span>
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="where-can-we-go-to-find-similar-sequences" class="level2" data-number="1.4">
<h2 data-number="1.4" class="anchored" data-anchor-id="where-can-we-go-to-find-similar-sequences"><span class="header-section-number">1.4</span> Where can we go to find similar sequences?</h2>
<p>There are many publicly-available biological sequence reference databases available. Each has different contents, and may focus on a particular aspect of biology.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>Always choose an appropriate sequence database suitable to answer your biological question.</p>
</div>
</div>
<section id="insdc" class="level3" data-number="1.4.1">
<h3 data-number="1.4.1" class="anchored" data-anchor-id="insdc"><span class="header-section-number">1.4.1</span> INSDC</h3>
<p>As a condition of publishing papers that describe new biological sequences - whether nucleotide or protein - those sequences must be deposited in a central public resource. For nucleotide sequence data, this is managed under the <a href="https://www.insdc.org/">International Sequence Database Collaboration</a> (INSDC), which is a collaboration between Japanese, US, and European organisations:</p>
<ul>
<li><a href="https://www.ddbj.nig.ac.jp/index-e.html">DDBJ</a> - the DNA Data Bank of Japan</li>
<li><a href="https://www.ebi.ac.uk/">EMBL-EBI</a> - the European Molecular Biology Laboratory - European Bioinformatics Institute</li>
<li><a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> - the National Center for Biotechnology Information, USA</li>
</ul>
<p>These organisations share their data, so have approximately the same core sequence records (although accession IDs vary between them).</p>
</section>
<section id="genbank" class="level3" data-number="1.4.2">
<h3 data-number="1.4.2" class="anchored" data-anchor-id="genbank"><span class="header-section-number">1.4.2</span> GenBank</h3>
<p>The <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a> database contains an annotated collection of all publicly available DNA sequences, and is maintained by NCBI as part of the INSDC. The initial release of GenBank in December 1982 contained 606 nucleotide sequences. As of December 2023, GenBank contains 249,060,436 nucleotide sequences, and 2,863,228,552 additional sequences deriving from whole-genome sequencing. The number of bases in GenBank <a href="https://www.ncbi.nlm.nih.gov/genbank/statistics/">doubles approximately every 18 months</a>.</p>
</section>
<section id="refseq" class="level3" data-number="1.4.3">
<h3 data-number="1.4.3" class="anchored" data-anchor-id="refseq"><span class="header-section-number">1.4.3</span> RefSeq</h3>
<p>NCBI also provides the <a href="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> (NCBI Reference Sequence) database, which provides a comprehensive, integrated, non-redundant, consistently annotated set of reference nucleotide, transcript, and protein sequences.</p>
</section>
<section id="uniprot" class="level3" data-number="1.4.4">
<h3 data-number="1.4.4" class="anchored" data-anchor-id="uniprot"><span class="header-section-number">1.4.4</span> UniProt</h3>
<p><a href="https://www.uniprot.org">UniProt</a> is a resource for protein sequence and annotation data. It comprises three databases:</p>
<ul>
<li><a href="https://www.uniprot.org/help/uniprotkb">UniProt Knowledgebase</a> (UniProtKB)</li>
<li><a href="https://www.uniprot.org/help/uniref">UniProt Reference Clusters</a> (UniRef)</li>
<li><a href="https://www.uniprot.org/help/uniparc">UniProt Archive</a> (UniParc)</li>
</ul>
<p>The main database of proteins is UniProtKB, and this is supported by UniRef - which clusters sequences on the basis of sequence identity - and UniParc, the archive of new, revised and obsolete sequences.</p>
<p>UniProtKB is itself comprised of two databases:</p>
<ul>
<li><a href="https://www.uniprot.org/uniprotkb?query=reviewed:true">Swiss-Prot</a>, a manually annotated and reviewed set of sequences where every annotation has been checked by a human</li>
<li><a href="">TrEMBL</a>, the remaining set of known sequences, whose annotations were dervied by automated assignment, and have not been reviewed by a human</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Manual annotation is time-consuming, and most proteins in UniProt have not been manually reviewed. As at 22nd February 2024:</p>
<ul>
<li>Swiss-Prot contains 570,830 manually-reviewed protein sequences</li>
<li>TrEMBL contains 249,751,891 automatically annotated protein sequences</li>
</ul>
<p>The current statistics can be found at <a href="https://www.uniprot.org/uniprotkb">UniProtKB</a></p>
</div>
</div>
</section>
<section id="interpro" class="level3" data-number="1.4.5">
<h3 data-number="1.4.5" class="anchored" data-anchor-id="interpro"><span class="header-section-number">1.4.5</span> InterPro</h3>
<p><a href="https://www.ebi.ac.uk/interpro/">Interpro</a> is a service providing functional analysis of proteins by querying their sequences against a set of <em>signature</em> databases that contain patterns characteristic of particular functions or protein structures (<a href="#fig-interpro" class="quarto-xref">Figure&nbsp;<span>1.4</span></a>).</p>
<div id="fig-interpro" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-interpro-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="assets/images/member_db.png" class="img-fluid figure-img">
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-interpro-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1.4: The <a href="https://www.ebi.ac.uk/interpro/">InterPro website</a> allows simultaneous sequence-based search of several signature databases, to help annotate and classify protein sequences.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="summary" class="level2" data-number="1.5">
<h2 data-number="1.5" class="anchored" data-anchor-id="summary"><span class="header-section-number">1.5</span> Summary</h2>
<div class="callout callout-style-default callout-tip callout-titled" title="What you know so far…">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
What you know so far…
</div>
</div>
<div class="callout-body-container callout-body">
<p>After reading through this section and following the links you should</p>
<ul>
<li>understand why it is important to be able identify similar sequences.</li>
<li>be able to name some of the major sequence databases that help us use sequence similarity to understand the possible origins and functions of biological sequences</li>
</ul>
<p>Now you are ready to start using some of the bioinformatic tools for determining sequence similarity.</p>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Olivella2013-rm" class="csl-entry" role="listitem">
Olivella, Mireia, Angel Gonzalez, Leonardo Pardo, and Xavier Deupi. 2013. <span>“Relation Between Sequence and Structure in Membrane Proteins.”</span> <em>Bioinformatics</em> 29 (13): 1589–92.
</div>
<div id="ref-Punta2008-gq" class="csl-entry" role="listitem">
Punta, Marco, and Yanay Ofran. 2008. <span>“The Rough Guide to in Silico Function Prediction, or How to Use Sequence and Structure Information to Predict Protein Function.”</span> <em>PLoS Comput. Biol.</em> 4 (10): e1000160.
</div>
<div id="ref-Zhou2019-ny" class="csl-entry" role="listitem">
Zhou, Naihui, Yuxiang Jiang, Timothy R Bergquist, Alexandra J Lee, Balint Z Kacsoh, Alex W Crocker, Kimberley A Lewis, et al. 2019. <span>“The <span>CAFA</span> Challenge Reports Improved Protein Function Prediction and New Functional Annotations for Hundreds of Genes Through Experimental Screens.”</span> <em>Genome Biol.</em> 20 (1): 244.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Preface to {{< var ay >}} BM214 Workshop 2: Sequence Searching">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Preface to 2023-24 BM214 Workshop 2: Sequence Searching</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./pairwise.html" class="pagination-link" aria-label="Pairwise Sequence Alignment">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Pairwise Sequence Alignment</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>This workshop was written by Dr Leighton Pritchard and Dr Morgan Feeney</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sipbs-compbiol/BM214-Workshop-2/edit/main/intro.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM214-Workshop-2/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>